assets/src/vo/currency.js

```diff
@@ -0,0 +1,239 @@
+		this.thousandsSeparator = currencyConfig.thousandsSeparator || this.thousandsSeparator;
+		this.subunits = currencyConfig.subunits ||
+			Math.pow( 10, this.decimalMark );
+		Object.freeze( this );
+	}
+
+	/**
+	 * Returns the currency properties as an object formatted for the
+	 * accounting-js library configuration.
+	 * @return {{}}  An object shaped for what the accounting-js library expects
+	 */
+	toAccountingSettings() {
+		const decimalInfo = {
+			decimal: this.decimalMark,
+			thousand: this.thousandsSeparator,
+			precision: this.decimalPlaces,
+		};
+		return {
+			currency: {
+				symbol: this.sign,
+				format: {
+					pos: this.signB4 ? '%s%v' : '%v%s',
+					neg: this.signB4 ? '- $s%v' : '- %v%s',
+					zero: this.signB4 ? '%s--' : '--%s',
+				},
+				...decimalInfo,
+			},
+			number: decimalInfo,
+		};
+	}
+
+	/**
+	 * Returns JSON representation of this object.
+	 * @return {Object} Function returning the object to be serialized by
+	 * JSON.stringify
+	 */
+	toJSON() {
+		return {
+			code: this.code,
+			singularLabel: this.singularLabel,
+			pluralLabel: this.pluralLabel,
+			sign: this.sign,
+			signB4: this.signB4,
+			decimalMark: this.decimalMark,
+			thousandsSeparator: this.thousandsSeparator,
+			subunits: this.subunits,
+			decimalPlaces: this.decimalPlaces,
+		};
+	}
+
+	/**
+	 * This validates whether the passed in config has the required properties
+	 * (and correct types) for constructing a Currency object.
+	 *
+	 * @param {{}} config
+	 * @throws {Exception}
+	 * @throws {TypeError}
+	 */
+	static validateCurrencyConfig = ( config ) => {
+		if ( isEmpty( config ) ) {
+			throw new Exception(
+				'The configuration object provided to Currency must not' +
+				' be empty'
+			);
+		}
+		if ( ! config.code || ! isString( config.code ) ) {
+			throw new TypeError(
+				'The configuration object provided to Currency must have ' +
+				'a "code" property that is a string.'
+			);
+		}
+
+		if ( ! config.sign || ! isString( config.sign ) ) {
+			throw new TypeError(
+				'The configuration object provided to Currency must have a ' +
+				'"sign" property that is a string.'
+			);
+		}
+
+		if ( config.singularLabel && ! isString( config.singularLabel ) ) {
+			throw new TypeError(
+				'The singularLabel property on the configuration object ' +
+				'must be a string primitive.'
+			);
+		}
+
+		if ( config.pluralLabel && ! isString( config.pluralLabel ) ) {
+			throw new TypeError(
+				'The pluralLabel property on the configuration object ' +
+				'must be a string primitive.'
+			);
+		}
+
+		if ( config.signB4 && ! isBoolean( config.signB4 ) ) {
+			throw new TypeError(
+				'The signB4 property on the configuration object ' +
+				'must be a boolean primitive.'
+			);
+		}
+
+		if ( config.decimalPlaces && ! isNumber( config.decimalPlaces ) ) {
+			throw new TypeError(
+				'The decimalPlaces property on the configuration object ' +
+				'must be a number primitive'
+			);
+		}
```

Comment:
No.  There's actually a country that has currency with 4 decimal places.  However, I think we should leave the range not checked because of the advent of crypto currencies (which can have multiple decimal places.  See [chart here](https://en.wikipedia.org/wiki/ISO_4217).

Reply:
ooo good point about cryptos... hadn't considered that which is not a wise approach these days...